### PR TITLE
[BugFix] Filter OECD data using `start_date` and `end_date` parameters

### PR DESCRIPTION
**Pull request by the-praxs**
# [BugFix] Filter OECD data using `start_date` and `end_date` parameters

1. **Why**?

    To enable the functionality of filtering data using the previously defined standard parameters.

2. **What**?

    - Move constant variables in the provider models to `utils/constants.py`

    - Filter data using `start_date` and `end_date` through `filter` and `lambda` functions

3. **Impact**

    - Minimal
    
    - Impact Analysis breakdown:
        - _UX_ = 2
        - _Performance_ = 1
        - _Compatibility_ = 0
        - _Security_ = 0
        - _Scalability_ = 1
        - _Maintenance_ = 1
        - _Compliance_ = 0
        - _Dependency_ = 0

    - **Total Impact Score** = 5

4. **Testing Done**:

    Tested on the notebook with and without the parameters.

### CODE REVIEW

File: openbb_platform/providers/oecd/openbb_oecd/models/long_term_interest_rate.py [hunk 1]
```diff
@@ -138,24 +90,30 @@ def extract_data(
...
+        country = "" if query.country == "all" else COUNTRY_TO_CODE_IR[query.country]
         url = "https://sdmx.oecd.org/public/rest/data/OECD.SDD.STES,DSD_KEI@DF_KEI,4.0/..IRLT...."
         data = helpers.get_possibly_cached_data(
             url, function="economy_long_term_interest_rate"
         )
-        query = f"FREQ=='{frequency}'"
-        query = query + f" & REF_AREA=='{country}'" if country else query
+        url_query = f"FREQ=='{frequency}'"
+        url_query = url_query + f" & REF_AREA=='{country}'" if country else url_query
         # Filter down
```
Comment (deeleeramone) [inline openbb_platform/providers/oecd/openbb_oecd/models/long_term_interest_rate.py:100]: So here, we would normally want these sections in the "transform_data", correct?

**Comment by deeleeramone** [comment]
Looks like the `read timeout` band-aid didn't do the trick :(

```
obb.economy.composite_leading_indicator()
```

```
OpenBBError: HTTPSConnectionPool(host='sdmx.oecd.org', port=443): Read timed out. (read timeout=30)
```

**Comment by jmaslek** [comment]
> Looks like the `read timeout` band-aid didn't do the trick :(
> 
> 
> 
> ```
> 
> obb.economy.composite_leading_indicator()
> 
> ```
> 
> 
> 
> ```
> 
> OpenBBError: HTTPSConnectionPool(host='sdmx.oecd.org', port=443): Read timed out. (read timeout=30)
> 
> ```
> 
> 
> 
> Even opening the URL in a browser is doing spinning wheel, so maybe there is a better way to query this API efficiently?
> 
> 
> 
> "https://sdmx.oecd.org/public/rest/data/OECD.SDD.STES,DSD_KEI@DF_KEI,4.0/..IRLT...."
> 
> 
> 
> Either way, can't really test this PR without the HTTP request working.

I was gonna venture into filtering the query down because the full data is just too much

**Comment by the-praxs** [comment]
> > Looks like the `read timeout` band-aid didn't do the trick :(
> > 
> > 
> > 
> > ```
> > 
> > obb.economy.composite_leading_indicator()
> > 
> > ```
> > 
> > 
> > 
> > ```
> > 
> > OpenBBError: HTTPSConnectionPool(host='sdmx.oecd.org', port=443): Read timed out. (read timeout=30)
> > 
> > ```
> > 
> > 
> > 
> > Even opening the URL in a browser is doing spinning wheel, so maybe there is a better way to query this API efficiently?
> > 
> > 
> > 
> > "https://sdmx.oecd.org/public/rest/data/OECD.SDD.STES,DSD_KEI@DF_KEI,4.0/..IRLT...."
> > 
> > 
> > 
> > Either way, can't really test this PR without the HTTP request working.
> 
> I was gonna venture into filtering the query down because the full data is just too much

Will the last 10 years of data be enough when no dates are provided?

**Comment by deeleeramone** [comment]
> > > Looks like the `read timeout` band-aid didn't do the trick :(
> > > ```
> > > 
> > > obb.economy.composite_leading_indicator()
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > ```
> > > 
> > > OpenBBError: HTTPSConnectionPool(host='sdmx.oecd.org', port=443): Read timed out. (read timeout=30)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Even opening the URL in a browser is doing spinning wheel, so maybe there is a better way to query this API efficiently?
> > > "https://sdmx.oecd.org/public/rest/data/OECD.SDD.STES,DSD_KEI@DF_KEI,4.0/..IRLT...."
> > > Either way, can't really test this PR without the HTTP request working.
> > 
> > 
> > I was gonna venture into filtering the query down because the full data is just too much
> 
> Will the last 10 years of data be enough when no dates are provided?

Definitely not. Getting a complete time series is what we are trying to do.

**Comment by the-praxs** [comment]
> > > > Looks like the `read timeout` band-aid didn't do the trick :(
> > > > ```
> > > > 
> > > > obb.economy.composite_leading_indicator()
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > ```
> > > > 
> > > > OpenBBError: HTTPSConnectionPool(host='sdmx.oecd.org', port=443): Read timed out. (read timeout=30)
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > Even opening the URL in a browser is doing spinning wheel, so maybe there is a better way to query this API efficiently?
> > > > "https://sdmx.oecd.org/public/rest/data/OECD.SDD.STES,DSD_KEI@DF_KEI,4.0/..IRLT...."
> > > > Either way, can't really test this PR without the HTTP request working.
> > > 
> > > 
> > > I was gonna venture into filtering the query down because the full data is just too much
> > 
> > 
> > Will the last 10 years of data be enough when no dates are provided?
> 
> Definitely not. Getting a complete time series is what we are trying to do.

I have never encountered the timeout for any of the functions here. Is there a way to reproduce it?

**Review by jmaslek — APPROVED** [review]
I'm going to approve this to move forward so I can edit the large requests

**Pull request merged**